region/src/main/java/org/motechproject/nms/region/csv/impl/CircleImportServiceImpl.java

```diff
@@ -0,0 +1,89 @@
+package org.motechproject.nms.region.csv.impl;
+
+import org.motechproject.nms.csv.exception.CsvImportDataException;
+import org.motechproject.nms.csv.utils.ConstraintViolationUtils;
+import org.motechproject.nms.csv.utils.CsvImporterBuilder;
+import org.motechproject.nms.csv.utils.CsvMapImporter;
+import org.motechproject.nms.csv.utils.GetString;
+import org.motechproject.nms.region.csv.CircleImportService;
+import org.motechproject.nms.region.domain.Circle;
+import org.motechproject.nms.region.domain.State;
+import org.motechproject.nms.region.repository.CircleDataService;
+import org.motechproject.nms.region.repository.StateDataService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.supercsv.cellprocessor.ift.CellProcessor;
+
+import javax.validation.ConstraintViolationException;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.HashMap;
+import java.util.Map;
+
+@Service("circleImportService")
+public class CircleImportServiceImpl implements CircleImportService {
+    public static final String CIRCLE_NAME = "Circle";
+    public static final String STATE_NAME = "State";
+
+    private StateDataService stateDataService;
+    private CircleDataService circleDataService;
+
+    @Autowired
+    public CircleImportServiceImpl(StateDataService stateDataService, CircleDataService circleDataService) {
+        this.stateDataService = stateDataService;
+        this.circleDataService = circleDataService;
+    }
+
+    @Override
+    @Transactional
+    public void importData(Reader reader) throws IOException {
+        CsvMapImporter csvImporter = new CsvImporterBuilder()
+                .setProcessorMapping(getProcessorMapping())
+                .createAndOpen(reader);
+        Map<String, Object> record;
+        while (null != (record = csvImporter.read())) {
+            try {
+                String circleName = (String) record.get(CIRCLE_NAME);
+                if (circleName == null) {
+                    throw new CsvImportDataException(String.format("CSV import error, circle missing on line %s",
```

Comment:
Probably don't want to include "CSV import error, " since you're already saying it on line 77
Same comment below as well...